homeassistant/components/livisi/__init__.py

```diff
@@ -0,0 +1,60 @@
+"""The Livisi Smart Home integration."""
+from __future__ import annotations
+
+import asyncio
+from typing import Final
+
+from aiolivisi import AioLivisi
+
+from homeassistant import core
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client, device_registry as dr
+
+from .const import DOMAIN, SWITCH_PLATFORM
+from .coordinator import LivisiDataUpdateCoordinator
+
+PLATFORMS: Final = [SWITCH_PLATFORM]
+
+
+async def async_setup_entry(hass: core.HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Livisi Smart Home from a config entry."""
+    web_session = aiohttp_client.async_get_clientsession(hass)
+    aiolivisi = AioLivisi(web_session)
+    coordinator = LivisiDataUpdateCoordinator(hass, entry, aiolivisi)
+    await coordinator.async_setup()
+    await coordinator.async_set_all_rooms()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+    if (
+        coordinator.serial_number is None
+        or coordinator.controller_type is None
+        or coordinator.os_version is None
+    ):
+        return False
```

Comment:
What is the reason for the missing data?

Reply:
I removed this.